frontend/apps/reader/modules/readerhyphenation.lua

```diff
@@ -100,22 +113,29 @@ function ReaderHyphenation:onPreRenderDocument(config)
     -- This is called after the document has been loaded
     -- so we can use the document language.
 
+    -- Use the one manually set for this document
     local hyph_alg = config:readSetting("hyph_alg")
-    if not hyph_alg then
+    if not hyph_alg then -- If none, use the one manually set as default (with Hold)
+        hyph_alg = G_reader_settings:readSetting("hyph_alg_default")
+    end
+    if not hyph_alg then -- If none, use the one associated with document's language
         hyph_alg = self:getDictForLanguage(self.ui.document:getProps().language)
     end
-    if not hyph_alg then
+    if not hyph_alg then -- If none, use the one manually set as fallback (with Hold)
         hyph_alg = G_reader_settings:readSetting("hyph_alg_fallback")
     end
     if hyph_alg then
         self.ui.document:setHyphDictionary(hyph_alg)
     end
+    -- If we haven't set any, hardcoded English_US_hyphen_(Alan).pdb (in cre.cpp) will be used
     self.hyph_alg = cre.getSelectedHyphDict()
 end
 
-function ReaderHyphenation:onSaveSettings()
-    self.ui.doc_settings:saveSetting("hyph_alg", self.hyph_alg)
-end
+-- We only save hyph_alg when it is manually set, so
+-- that new global fallback or default can be used.
+-- function ReaderHyphenation:onSaveSettings()
+--     self.ui.doc_settings:saveSetting("hyph_alg", self.hyph_alg)
```

Comment:
But with this, the self.hyph_alg (which is set when one is chosed with default/book language/fallback) is saved as per-document setting. So, later, if one change default or fallback, the previous default or fallback which had been saved that way will be used instead of any new default/fallback.
I moved the saveSetting at the top, in the (non-hold) callback, so it's saved only when another hyph_alg (than the default/fallback) is chosen.
Makes more sense to me, that getting stuck to some old badly chose default/fallback/original book language.